staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher_test.go

```diff
@@ -353,8 +353,13 @@ func TestProgressNotify(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Watch failed: %v", err)
 	}
-	result := &example.Pod{ObjectMeta: metav1.ObjectMeta{ResourceVersion: out.ResourceVersion}}
-	testCheckResult(t, watch.Bookmark, w, result)
+	testCheckResultFunc(t, watch.Bookmark, w, func(object runtime.Object) error {
+		pod, ok := object.(*example.Pod)
+		if !ok {
+			return fmt.Errorf("got %T, not *example.Pod", object)
+		}
+		return resourceVersionNoLaterThan(out.ResourceVersion)(pod.ResourceVersion)
```

Comment:
(as an aside, I find the updated version harder to understand... not sure if there's a clearer way to express this)